### DISCUSSION TITLE
What do the settings for Maximum Elevation and Minimum Elevation do? where do I get them from?

### DISCUSSION THREAD

**Discussion by lennon101** [basbruss/adaptive-cover#213] [Q&A] [answered]
I don't really understand where i'm supposed to get the values for the maximum and minimum elevation from? Is this relating to how close I am to the equator and how high the sun rises relative to the bottom of the horizon? And if so, won't this value change between summer and winter? What I mean by this is: in winter in Australia the sun doesn't get as high in the sky as it does in summer. Am I thinking about this the right way or is this setting relating to something else? Thanks for the help.

**Comment by basbruss** [comment]
These parameters are optional and are only needed if you have objects that naturally block out the sun. For instance, if you have a building or apartment block in front of the window, you can adjust the minimum elevation degree to a higher value to keep the covers open while the sun is still climbing behind the object.

The maximum value can be used when you have an eave or similar structure, allowing the covers to remain open after the sun reaches a high enough elevation to be fully blocked by the eave.

**Reply by lennon101** [reply]
ah ok that makes perfect sense! and thanks so much for the quick reply. So just to clarify, I do in fact have eaves that stick out over the window. How would I determine what the maximum elevation is of the sun? I see that the units those parameters are expecting is in degrees. What angle am I measuring exactly? An explanation of "from x to y" would be great! 

Thanks so much!

**Reply by basbruss** [reply, marked as answer]
![sun elevation](https://github.com/basbruss/adaptive-cover/assets/68892092/e2503da5-fa45-469d-8a8a-4dfefcd383f9)

Based on the illustration above the wanted angle is that of `B`, which corresponds to the elevation degree of the sun.
Using some geometric formulas we can calculate `B` based on the lengths of sides `b` and `c`.

- side b is the distance from the lowest point of the eave to the bottom of the window/cover
- side c is the horizontal distance from the eave to the cover + the distance to the work area set in the integration (default is 0.5 meters)

You can put both measurements into a calculator https://www.calculator.net/triangle-calculator.html and fill in the values of sides `b` and `c`, plus add an angle of `90` degrees to angle `A`, since that corner is always perpendicular. That will give you the angle `B` also result, which in turn is the maximum elevation angle you need to worry about.

**Reply by lennon101** [reply]
Amazing. Thanks so much for this! If it isn't already, I recommed putting this into the readme! Also wanted to say: incredible work on this integration. Can I shout you a coffee or something? I'd love to show my appreciation and support. cheers!

**Comment by lloydsmart** [comment]
@basbruss I think I have a related question. I've configured your integration for my office blind at home, and am loving it so far, but I find that the blinds reach 100% closed well before it gets dark. What I'd like is for them to block the glare in the afternoon as they currently do, but then once the sun sets below the horizon to fully open until it actually gets dark.

Here's the view from my window, facing 296°:
![window](https://github.com/user-attachments/assets/453bc4cf-6d0c-45bf-9b95-ad3572bbee70)

I think because there's a slight upward hill between my window and the horizon (or maybe it's just all the foliage in the field), the sun is invisible before the integration thinks it's fully "set".

Is this a situation where I'd use the "Minimum elevation of the sun" setting? If so, how would you recommend I calculate the appropriate angle?

Thanks for all your work on this.

**Reply by basbruss** [reply]
No need to calculate the angle. You can observe when the light is not coming in directly and take the elevation value from the sun.sun entity at that time.

**Reply by lloydsmart** [reply]
Wonderful, thanks for the pointer!

I'm also looking in to setting up a blind spot for that tree.

Starting to get a feel for this now.
